### PR TITLE
Feature/separate cmd exec

### PR DESCRIPTION
**Pull request by lebdron**
## What is this pull request?
Move `validate()` and `execute()` from `model::Command` to separate `CommandExecutor` classes.
This class is used in `TemporaryWsv` and `MutableStorage`, so that their users don't have to manually call `validate` and `execute`, requiring only to specify custom precondition for `Transaction` and `Block` respectively
   
## Why do you implement it? Why do we need this pull request?
Model depended on ametsuchi, which is not correct.
  
## How to use the features provided in the pull request?
Instantiate `CommandExecutorFactory`, use `getCommandExecutor()` for your commands.

### CODE REVIEW

File: test/module/irohad/model/command_validate_execute_test.cpp [hunk 1]
```diff
@@ -41,6 +42,8 @@ class CommandValidateExecuteTest : public ::testing::Test {
  public:
 
   void SetUp() override {
+    factory = CommandExecutorFactory::create().value();
```
Comment (muratovv) [inline test/module/irohad/model/command_validate_execute_test.cpp]: Add tests for `AddDomain` and `AddPeer`.

File: irohad/ametsuchi/impl/test_storage_impl.cpp [hunk 2]
```diff
@@ -68,23 +68,9 @@ namespace iroha {
...
-                                           if (not command->execute(query,
-                                                                    executor)) {
-                                             return false;
-                                           }
-                                         }
-                                       }
-                                       return true;
-                                     });
+      auto inserted =
+          storage->apply(block, [](const auto &current_block, auto &query,
```
Comment (lebdron) [inline irohad/ametsuchi/impl/test_storage_impl.cpp]: This requires an extensive discussion. This PR tries to move logic from command without breaking other components much

File: test/module/irohad/model/command_validate_execute_test.cpp (hunk 1 above)
Comment (lebdron) [inline test/module/irohad/model/command_validate_execute_test.cpp]: https://github.com/hyperledger/iroha/pull/518/files#diff-9ba5384c50df125ef0eea704a8963763R442

File: irohad/validation/impl/stateful_validator_impl.cpp [hunk 3]
```diff
@@ -30,29 +30,20 @@ namespace iroha {
...
         auto account = temporaryWsv.getAccount(tx.creator_account_id);
         // Check if tx creator has account and has quorum to execute transaction
         if (!account || tx.signatures.size() < account.value().quorum)
           return false;
 
         // Check if signatures in transaction are account signatory
         auto account_signs = temporaryWsv.getSignatories(tx.creator_account_id);
-        if (!account_signs)
+        if (not account_signs)
           // No signatories found
```
Comment (grimadas) [inline irohad/validation/impl/stateful_validator_impl.cpp]: Maybe we can add logger here  ?

**Comment by grimadas** [comment]
Also I think it is reasonable to add logger to impl/command_executor.cpp where appropriate (there are comments in there)

File: irohad/validation/impl/stateful_validator_impl.cpp (hunk 3 above)
Comment (neewy) [inline irohad/validation/impl/stateful_validator_impl.cpp]: I suggest creating a separate issue for this task and merge this code, if it looks good @grimadas @muratovv

**Pull request merged**